apollo-router/src/plugins/telemetry/tracing/datadog.rs

```diff
@@ -0,0 +1,76 @@
+use crate::plugins::telemetry::config::Trace;
+use crate::plugins::telemetry::{GenericWith, TracingConfigurator};
+use opentelemetry::sdk::trace::Builder;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use tower::BoxError;
+use url::Url;
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    pub endpoint: AgentEndpoint,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub enum AgentEndpoint {
+    Default(AgentDefault),
+    Url(Url),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+pub enum AgentDefault {
+    Default,
+}
+
+impl TracingConfigurator for Config {
+    fn apply(&self, builder: Builder, trace_config: &Trace) -> Result<Builder, BoxError> {
+        let url = match &self.endpoint {
+            AgentEndpoint::Default(_) => None,
+            AgentEndpoint::Url(s) => Some(s),
+        };
```

Comment:
That could be a method on `AgentEndpoint` as we use exactly the same match everywhere